### ISSUE TITLE
Single use CUP launchers blacklist by arsenal

### ISSUE THREAD

**Issue by JoramD0** [KillahPotatoes/KP-Liberation#707]
### Basic Information

**Mission version:** v0.96.6

**Map used:** Altis

**Mods used:** CUP, ACE & N/A

**Dedicated or local game:** Dedicated

**Source of the mission file (URL):** https://github.com/KillahPotatoes/KP-Liberation/releases

### Individual Things
**Did you edit anything within the mission files?**
Yes

**If yes, what did you edit exactly?**
Config, whitelist & mission_params

### Short Issue Description
Single use CUP launchers are not allowed by the arsenal. 

### Expected Behaviour

The launchers should be allowed.

### Steps to Reproduce the Issue

Try to save/load a loadout with a single use CUP launcher.

### Screenshots, Serverlog or any other helpful Information
My guess is that they changed something in one of the latest CUPdates, it should be fixed by adding the classnames of the "loaded" variants to the `allowed_items_extension` array.
I did this but have not been able to test it yet.
These are (from what I can tell) the new classnames:
   ``` "CUP_launch_M72A6_Loaded",
    "CUP_launch_M72A6_Special_Loaded",
    "CUP_launch_Igla_Loaded",
    "CUP_launch_M136_Loaded",
    "CUP_launch_NLAW_Loaded",
    "CUP_launch_RPG18_Loaded",
    "CUP_launch_FIM92Stinger_Loaded",
    "CUP_launch_9K32Strela_Loaded"```

**Comment by JoramD0** [comment]
Adding the `_loaded` variants to the `allowed_items_extension` array only duplicates it. Looking at the config for the launchers the `_loaded` variant should be obtainable (scope=1) but shouldn't show up in the arsenal. I don't know why the arsenal is removing it even though it is set to use the blacklist

**Comment by Wyqer** [comment]
Attach your config file you've setup with these in the comment and also a server rpt where you get these weapons removed. It should state the exact classnames in the log.

**Comment by JoramD0** [comment]
Here are the requested files, in the log it says the item is blacklisted but it is not in the blacklist.

https://gist.github.com/JoramD0/6d783c9f661d79674c343844b5e9619f

**Comment by Wyqer** [comment]
You should add it to the extension, maybe @Eogos could also have a look at this, as he uses CUP mostly for Liberation while I never played really with CUP.
Basically with adding them to the extension it should be fine.

**Comment by JoramD0** [comment]
Yea, I tested that and it shows the items twice even though the config scope is set to 1. Almost seems like items with scope=1 set are automatically blacklisted

**Comment by Eogos** [comment]
I'll do some testing with it later and see what I can dig up.

**Comment by JoramD0** [comment]
After playing some more I found other items being removed. For example scopes with 2 states (like a magnifier in front of the scope). Usually these are made by making 2 scopes and then assigning a function/key to switch between the 2 models. And then setting the scope of the default variant to 2 and the other one set to scope 1 (only accessible from script).

Is it possible the blacklist function is blocking all items that have their config set to `scope=1;`?
I tried looking at the code for it but I cant make much sense of it.

**Comment by Wyqer** [comment]
> Is it possible the blacklist function is blocking all items that have their config set to scope=1;?

It does indeed, good catch.
Haven't looked until now about this. Especially as it's a really old function which is used and implemented years ago in the framework.

As we're currently converting all functions there to a proper CfgFunctions library, we'll very likely tweak this also. If you've also fun with improving functions of really old and horrible code, feel free to grab some functions and create a PR with the refactor/simplification ^^

Ref for current fetching function:
https://github.com/KillahPotatoes/KP-Liberation/blob/master/Missionframework/scripts/shared/functions/LARs_fnc_addAllVirtualCargo.sqf#L24

Ref for current function refactoring for 0.96.7:
https://github.com/KillahPotatoes/KP-Liberation/tree/v0.96.7-CfgFunctions/Missionframework/functions

**Comment by veteran29** [comment]
Fixed in #718, handling of CBA disposables will be added with 0.96.7 release.
